test/higher_order_ops/test_invoke_subgraph.py

```diff
@@ -391,6 +391,53 @@ def fn(x, y):
         ):
             opt_fn(x, y)
 
+    def test_ac(self):
+        def fn1(x):
+            return torch.cos(x)
+
+        def fn1_checkpoint(x):
+            return (torch.utils.checkpoint.checkpoint(fn1, x, use_reentrant=False),)
+
+        def fn2(x):
+            return torch.sin(x)
+
+        def fn2_checkpoint(x):
+            return (torch.utils.checkpoint.checkpoint(fn2, x, use_reentrant=False),)
```

Comment:
I don't remember how AC with SAC works, but that is probably not cacheable because we do not trace out the SAC function

Reply:
So action item here: can we test that checkpoint with SAC is not cacheable?

Reply:
I think SAC also works because the context_fn is sort of hackily passed on to aot_autograd. I added a test,it passes. But I can look more if this does not make sense.